internal/tiltfile/v1alpha1/types.go

```diff
@@ -73,3 +79,111 @@ func (p Plugin) extensionRepo(t *starlark.Thread, fn *starlark.Builtin, args sta
 	obj.ObjectMeta.Annotations = annotations
 	return p.register(t, obj)
 }
+
+func (p Plugin) fileWatch(t *starlark.Thread, fn *starlark.Builtin, args starlark.Tuple, kwargs []starlark.Tuple) (starlark.Value, error) {
+	obj := &v1alpha1.FileWatch{
+		ObjectMeta: metav1.ObjectMeta{},
+		Spec:       v1alpha1.FileWatchSpec{},
+	}
+	var watchedPaths value.LocalPathList = value.NewLocalPathListUnpacker(t)
+	var ignores IgnoreDefList = IgnoreDefList{t: t}
+	var labels value.StringStringMap
+	var annotations value.StringStringMap
+	err := starkit.UnpackArgs(t, fn.Name(), args, kwargs,
+		"name", &obj.ObjectMeta.Name,
+		"labels?", &labels,
+		"annotations?", &annotations,
+		"watched_paths?", &watchedPaths,
+		"ignores?", &ignores,
+	)
+	if err != nil {
+		return nil, err
+	}
+
+	obj.Spec.WatchedPaths = watchedPaths.Value
+	obj.Spec.Ignores = ignores.Value
+	obj.ObjectMeta.Labels = labels
+	obj.ObjectMeta.Annotations = annotations
+	return p.register(t, obj)
+}
+
+type IgnoreDef struct {
+	*starlark.Dict
+	Value v1alpha1.IgnoreDef
+	t     *starlark.Thread // instantiation thread for computing abspath
+}
+
+func (o *IgnoreDef) Unpack(v starlark.Value) error {
+	obj := v1alpha1.IgnoreDef{}
+
+	mapObj, ok := v.(*starlark.Dict)
```

Comment:
Did you consider having `tilt-starlark-codegen` make struct types for nested complex types vs `dict`s?

Reply:
i'm also going to add a built-in v1alpha1.ignore_def() that returns this object

re: structs vs dicts - i think it would be totally reasonable to accept both structs and dicts here. I started with `dict` because i expect python programmers to be more familar with it (as opposed to struct, which is starlark-only). i don't think it's too hard to convert between them.

https://docs.bazel.build/versions/main/skylark/lib/struct.html